### PR TITLE
[MOD] Modularizes app into an app and an activity module

### PR DESCRIPTION
**Pull request by keyboardsurfer**
This PR contains the first step to full modularization.
All activities but the HomeActivity are now in the activities module, together with many resources that are required just by these activities.

### CODE REVIEW

**Comment by keyboardsurfer** [comment]
Fixes #272 and #281

File: app/src/main/java/io/plaidapp/ui/FeedAdapter.java [hunk 1]
```diff
@@ -704,9 +707,9 @@ public void onMapSharedElements(List<String> names, Map<String, View> sharedElem
 
     static class DesignerNewsStoryHolder extends RecyclerView.ViewHolder implements Divided {
 
-        @BindView(R.id.story_title) BaselineGridTextView title;
-        @BindView(R.id.story_comments) TextView comments;
-        @BindView(R.id.pocket) ImageButton pocket;
+        BaselineGridTextView title;
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/FeedAdapter.java]: won't this crash once DesignerNews displays data? 
i think we need to call findViewById, and remove `ButterKnife.bind(this, itemView);`

Comment (florina-muntenescu) [reply app/src/main/java/io/plaidapp/ui/FeedAdapter.java]: How come ProductHunt below works with ButterKnife?

File: app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java [hunk 2]
```diff
@@ -57,17 +58,17 @@ public boolean canReuseUpdatedViewHolder(@NonNull RecyclerView.ViewHolder viewHo
...
     }
 
     @NonNull
     @Override
-    public ItemHolderInfo recordPreLayoutInformation(@NonNull RecyclerView.State state,
-                                                     @NonNull RecyclerView.ViewHolder viewHolder,
-                                                     int changeFlags,
-                                                     @NonNull List<Object> payloads) {
-        ItemHolderInfo info =
+    public RecyclerView.ItemAnimator.ItemHolderInfo recordPreLayoutInformation(@NonNull RecyclerView.State state,
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java]: line too long. move the params to the next line.

File: app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java [hunk 3]
```diff
@@ -57,17 +58,17 @@ public boolean canReuseUpdatedViewHolder(@NonNull RecyclerView.ViewHolder viewHo
     }
 
     @Override
-    public ItemHolderInfo obtainHolderInfo() {
+    public RecyclerView.ItemAnimator.ItemHolderInfo obtainHolderInfo() {
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java]: can you import `RecyclerView.ItemAnimator`?

File: app/src/main/java/io/plaidapp/ui/widget/FourThreeImageView.java [hunk 4]
```diff
@@ -30,8 +32,8 @@ public FourThreeImageView(Context context, AttributeSet attrs) {
 
     @Override
     protected void onMeasure(int widthSpec, int heightSpec) {
-        int fourThreeHeight = MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthSpec) * 3 / 4,
-                MeasureSpec.EXACTLY);
+        int fourThreeHeight = View.MeasureSpec.makeMeasureSpec(View.MeasureSpec.getSize(widthSpec) * 3 / 4,
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/widget/FourThreeImageView.java]: Can you import `View.MeasureSpec`?

File: app/src/main/java/io/plaidapp/ui/widget/FourThreeLinearLayout.java [hunk 5]
```diff
@@ -30,8 +32,8 @@ public FourThreeLinearLayout(Context context, AttributeSet attrs) {
 
     @Override
     protected void onMeasure(int widthSpec, int heightSpec) {
-        int fourThreeHeight = MeasureSpec.makeMeasureSpec(MeasureSpec.getSize(widthSpec) * 3 / 4,
-                MeasureSpec.EXACTLY);
+        int fourThreeHeight = View.MeasureSpec.makeMeasureSpec(View.MeasureSpec.getSize(widthSpec) * 3 / 4,
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/ui/widget/FourThreeLinearLayout.java]: can you import `View.MeasureSpec`

File: app/src/main/java/io/plaidapp/util/ActivityHelper.kt [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.support.customtabs.CustomTabsIntent
+import android.support.customtabs.CustomTabsSession
+import android.support.v4.content.ContextCompat
+import io.plaidapp.R
+import io.plaidapp.data.api.designernews.UpvoteStoryService
+
```
Comment (florina-muntenescu) [inline app/src/main/java/io/plaidapp/util/ActivityHelper.kt]: Add a comment about the role of this file

File: app/src/main/res/values/styles.xml [hunk 7]
```diff
@@ -383,4 +128,20 @@
...
+    <style name="Plaid.Translucent.DesignerNewsLogin" />
+    <style name="Plaid.Translucent.PostDesignerNewsStory" />
+    <style name="Plaid.Translucent.Dribbble" />
+    <style name="Plaid.Translucent.Dribbble.Login" />
+    <style name="Plaid.Translucent.Dribbble.Shot" />
+    <style name="Plaid.Translucent.Dribbble.Player" />
+    <style name="Plaid.Translucent.Dribbble.PlayerSheet" />
+    <style name="Plaid.Translucent.Search" />
+    <style name="TextAppearance"/>
+    <style name="TextAppearance.EmptyText" />
```
Comment (florina-muntenescu) [inline app/src/main/res/values/styles.xml]: AS tells me `cannot resolve symbol TextAppearance` for this and the next 2

File: app/src/main/res/values/styles.xml [hunk 8]
```diff
@@ -56,108 +56,16 @@
         <item name="android:windowDrawsSystemBarBackgrounds">true</item>
     </style>
```
Comment (florina-muntenescu) [inline app/src/main/res/values/styles.xml]: ProductHunt shows incorrect size for the text, in the feed

File: app/src/main/java/io/plaidapp/ui/FeedAdapter.java (hunk 1 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/ui/FeedAdapter.java]: My bad. This is in the app module. Butterknife works here. Reverted.

File: app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java (hunk 3 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java]: Done

File: app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java (hunk 2 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/ui/HomeGridItemAnimator.java]: Done

File: app/src/main/java/io/plaidapp/ui/widget/FourThreeImageView.java (hunk 4 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/ui/widget/FourThreeImageView.java]: Done

File: app/src/main/java/io/plaidapp/ui/widget/FourThreeLinearLayout.java (hunk 5 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/ui/widget/FourThreeLinearLayout.java]: Done

File: app/src/main/java/io/plaidapp/util/ActivityHelper.kt (hunk 6 above)
Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/util/ActivityHelper.kt]: Done

File: app/src/main/res/values/styles.xml (hunk 8 above)
Comment (keyboardsurfer) [reply app/src/main/res/values/styles.xml]: My refactoring was a bit too greedy here. Fixed

File: app/src/main/res/values/styles.xml (hunk 7 above)
Comment (keyboardsurfer) [reply app/src/main/res/values/styles.xml]: Fixed

File: features/activities/src/main/java/io/plaidapp/ui/PostNewDesignerNewsStory.java [hunk 9]
```diff
@@ -141,6 +137,72 @@ public boolean onPreDraw() {
...
     }
 
+    private void bindResources() {
+        bottomSheet = findViewById(R.id.bottom_sheet);
+        bottomSheet.setOnClickListener(view -> dismiss());
+        bottomSheetContent = findViewById(R.id.bottom_sheet_content);
+        sheetTitle = findViewById(R.id.title);
+        scrollContainer = findViewById(R.id.scroll_container);
+        title = findViewById(R.id.new_story_title);
+        title.addTextChangedListener(new TextWatcher() {
```
Comment (florina-muntenescu) [inline features/activities/src/main/java/io/plaidapp/ui/PostNewDesignerNewsStory.java]: am I looking wrong? I don't see the addTextChangedListener in the before file

Comment (florina-muntenescu) [reply features/activities/src/main/java/io/plaidapp/ui/PostNewDesignerNewsStory.java]: yes I was.

**Pull request merged**